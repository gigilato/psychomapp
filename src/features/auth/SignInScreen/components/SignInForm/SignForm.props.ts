import { IBoxProps } from '$atoms'

export interface ISignInFormProps extends IBoxProps {}
