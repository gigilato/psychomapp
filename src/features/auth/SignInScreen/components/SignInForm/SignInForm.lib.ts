import { zodResolver } from '@hookform/resolvers/zod'
import { useRef } from 'react'
import { useForm } from 'react-hook-form'
import { TextInput } from 'react-native'
import { z } from 'zod'

import { signInAsync } from '$infra/auth'

export const SignInFormSchema = z
  .object({
    email: z.string().min(1, 'Champs requis').email('Email incorrect'),
    password: z.string().min(1, 'Champs requis'),
  })
  .required()
export type SignInFormType = z.infer<typeof SignInFormSchema>

export const useSignInForm = () => {
  const passwordRef = useRef<TextInput>(null)

  const form = useForm<SignInFormType>({
    resolver: zodResolver(SignInFormSchema),
    defaultValues: {
      email: '',
      password: '',
    },
    mode: 'onBlur',
  })

  const onPressSubmit = form.handleSubmit(({ email, password }) => {
    signInAsync(email, password)
  })

  return { passwordRef, form, onPressSubmit }
}
