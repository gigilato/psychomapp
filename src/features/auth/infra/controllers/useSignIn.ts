import { useMutation } from '@tanstack/react-query'

import { signInAsync } from '$infra/auth'
import { showToast } from '$infra/toast'

type Variables = {
  email: string
  password: string
}
export const useSignIn = () => {
  const signInMutation = useMutation({
    mutationFn: ({ email, password }: Variables) => {
      return signInAsync(email, password)
    },
    onError: () => {
      showToast('Connexion impossible, v√©rifiez vos identifiants.')
    },
  })

  return signInMutation
}
