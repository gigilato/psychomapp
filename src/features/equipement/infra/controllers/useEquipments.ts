import { useQuery } from '@tanstack/react-query'

import { getEquipments } from '../fetchers'

export const queryKey = ['equipments']
export const useEquipments = () => {
  const query = useQuery({
    queryKey,
    queryFn: async () => {
      return getEquipments()
    },
  })
  return query
}
