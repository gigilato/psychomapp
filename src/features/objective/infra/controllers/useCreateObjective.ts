import { useMutation, useQueryClient } from '@tanstack/react-query'

import { i18n } from '$infra/i18n'
import { showToast } from '$infra/toast'

import { queryKey } from './useObjectives'
import { CreateObjectiveVariables, createObjective, getObjective } from '../fetchers'

export const useCreateObjective = () => {
  const queryClient = useQueryClient()

  const mutation = useMutation({
    mutationFn: async (variables: CreateObjectiveVariables) => {
      const { result, data } = await createObjective(variables)
      if (result.status !== 201) throw new Error('CreateObjectiveError')
      const objective = await getObjective(data.id)
      return objective
    },
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey })
    },
    onError: () => {
      showToast(i18n.t('errors.default'))
    },
  })
  return mutation
}
