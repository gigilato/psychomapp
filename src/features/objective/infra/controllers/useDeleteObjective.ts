import { useMutation, useQueryClient } from '@tanstack/react-query'

import { i18n } from '$infra/i18n'
import { showToast } from '$infra/toast'
import { Objective } from '$types/database'

import { queryKey } from './useObjectives'
import { DeleteObjectiveVariables, deleteObjective } from '../fetchers'

export const useDeleteObjective = () => {
  const queryClient = useQueryClient()

  const mutation = useMutation({
    mutationFn: async (variables: DeleteObjectiveVariables) => {
      const result = await deleteObjective(variables.id)
      if (result.status !== 204) throw new Error('DeleteObjectiveError')
      return variables
    },
    onMutate: async (variables) => {
      await queryClient.cancelQueries({ queryKey })
      const previous = queryClient.getQueryData<Objective[]>(queryKey)
      queryClient.setQueryData<Objective[]>(
        queryKey,
        (current) => current?.filter((item) => item.id !== variables.id)
      )
      return { previous }
    },
    onError: (_, variables, context) => {
      if (context?.previous) queryClient.setQueryData<Objective[]>(queryKey, context.previous)
      showToast(i18n.t('errors.default'))
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey })
    },
  })
  return mutation
}
