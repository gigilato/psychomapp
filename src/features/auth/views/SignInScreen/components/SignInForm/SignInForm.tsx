import { Box } from '$atoms'
import { FormTextInput } from '$forms'
import { Button, PressableText } from '$molecules'

import { ISignInFormProps } from './SignForm.props'
import { SignInFormType, useSignInForm } from './SignInForm.lib'

export const SignInForm = (props: ISignInFormProps) => {
  const { passwordRef, form, onPressSubmit, isPending, onPressForgotPassword } = useSignInForm()
  return (
    <Box {...props}>
      <FormTextInput<SignInFormType>
        name="email"
        control={form.control}
        type="email"
        label="Email"
        placeholder="Votre email"
        textContentType="emailAddress"
        onSubmitEditing={() => passwordRef.current?.focus()}
      />
      <FormTextInput<SignInFormType>
        className="mt-s"
        name="password"
        control={form.control}
        type="password"
        label="Mot de passe"
        placeholder="Votre mot de passe"
        ref={passwordRef}
      />
      <PressableText
        variant="link"
        ID="forgotPassword/button"
        event="clic_forgot_password"
        className="mt-s self-end"
        onPress={onPressForgotPassword}
      >
        Mot de passe oubli√© ?
      </PressableText>
      <Button
        ID="signIn/button"
        event="clic_sign_in"
        title="Se connecter"
        onPress={onPressSubmit}
        className="mt-xl"
        isLoading={isPending}
      />
    </Box>
  )
}
