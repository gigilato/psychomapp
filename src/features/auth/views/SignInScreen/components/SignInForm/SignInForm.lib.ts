import { zodResolver } from '@hookform/resolvers/zod'
import { useRef } from 'react'
import { useForm } from 'react-hook-form'
import { TextInput } from 'react-native'
import { z } from 'zod'

import { useSignIn } from '$features/auth/infra/controllers/useSignIn'
import { emailSchema, stringSchema } from '$forms/libs/form.schemas'
import { resetPasswordAsync } from '$infra/auth'
import { showToast } from '$infra/toast'

export const SignInFormSchema = z
  .object({
    email: emailSchema,
    password: stringSchema,
  })
  .required()
export type SignInFormType = z.infer<typeof SignInFormSchema>

export const useSignInForm = () => {
  const passwordRef = useRef<TextInput>(null)

  const form = useForm<SignInFormType>({
    resolver: zodResolver(SignInFormSchema),
    defaultValues: {
      email: '',
      password: '',
    },
    mode: 'onBlur',
  })

  const { mutateAsync, isPending } = useSignIn()

  const onPressSubmit = form.handleSubmit((formData) => {
    mutateAsync(formData)
  })

  const onPressForgotPassword = () => {
    const { email } = form.getValues()
    const { success } = emailSchema.safeParse(email)
    if (!success) {
      showToast('Veuillez renseigner un email valide.')
      return
    }
    resetPasswordAsync(email)
    showToast('Un email de réinitialisation vous a été envoyé.')
  }

  return { passwordRef, form, onPressSubmit, isPending, onPressForgotPassword }
}
