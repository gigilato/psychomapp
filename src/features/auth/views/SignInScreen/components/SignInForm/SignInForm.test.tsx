import { render } from '@testing-library/react-native'

import { useSignIn } from '$features/auth/infra/controllers/useSignIn'

import { SignInForm } from './SignInForm'

jest.mock('$features/auth/infra/controllers/useSignIn', () => ({
  useSignIn: jest.fn(),
}))

describe('SignInForm', () => {
  it('should render the form correctly', () => {
    ;(useSignIn as jest.Mock).mockImplementation(() => ({
      mutateAsync: jest.fn(),
      isPending: false,
    }))
    const component = render(<SignInForm />)
    expect(component).toMatchSnapshot()
  })

  it('should render the form loading', () => {
    ;(useSignIn as jest.Mock).mockImplementation(() => ({
      mutateAsync: jest.fn(),
      isPending: true,
    }))
    const component = render(<SignInForm />)
    expect(component).toMatchSnapshot()
  })
})
