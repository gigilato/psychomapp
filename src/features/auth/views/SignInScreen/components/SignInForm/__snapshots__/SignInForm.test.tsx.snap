// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should render the form correctly 1`] = `
<View>
  <View>
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "lineHeight": 18,
          },
          {
            "marginBottom": 8,
          },
          {
            "color": "#3F434A",
          },
        ]
      }
    >
      Email
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
          },
          {
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "overflow": "hidden",
          },
          {
            "borderBottomColor": "#3F434A",
            "borderLeftColor": "#3F434A",
            "borderRightColor": "#3F434A",
            "borderTopColor": "#3F434A",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="email"
          className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[14px] leading-[18px]"
          cursorColor="#6C464F"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Votre email"
          selectionColor="#6C464F"
          testID="input/email"
          textContentType="emailAddress"
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="input/email/RightIcon"
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "lineHeight": 18,
          },
          {
            "marginBottom": 8,
          },
          {
            "color": "#3F434A",
          },
        ]
      }
    >
      Mot de passe
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
          },
          {
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "overflow": "hidden",
          },
          {
            "borderBottomColor": "#3F434A",
            "borderLeftColor": "#3F434A",
            "borderRightColor": "#3F434A",
            "borderTopColor": "#3F434A",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="password"
          className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[14px] leading-[18px]"
          cursorColor="#6C464F"
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Votre mot de passe"
          secureTextEntry={true}
          selectionColor="#6C464F"
          testID="input/password"
          textContentType="password"
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="input/password/RightIcon"
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    event="clic_forgot_password"
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "marginTop": 12,
        },
        {
          "alignSelf": "flex-end",
        },
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="forgotPassword/button"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#3F434A",
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Mot de passe oubli√© ?
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animation="scale"
    collapsable={false}
    event="clic_sign_in"
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        {
          "backgroundColor": "#6C464F",
        },
        {
          "height": 44,
        },
        {
          "marginTop": 32,
        },
        [
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
          undefined,
        ],
      ]
    }
    testID="signIn/button"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Bold",
          },
          {
            "fontSize": 16,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Se connecter
    </Text>
  </View>
</View>
`;

exports[`SignInForm should render the form loading 1`] = `
<View>
  <View>
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "lineHeight": 18,
          },
          {
            "marginBottom": 8,
          },
          {
            "color": "#3F434A",
          },
        ]
      }
    >
      Email
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
          },
          {
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "overflow": "hidden",
          },
          {
            "borderBottomColor": "#3F434A",
            "borderLeftColor": "#3F434A",
            "borderRightColor": "#3F434A",
            "borderTopColor": "#3F434A",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="email"
          className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[14px] leading-[18px]"
          cursorColor="#6C464F"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Votre email"
          selectionColor="#6C464F"
          testID="input/email"
          textContentType="emailAddress"
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="input/email/RightIcon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 24,
              },
              [
                {
                  "marginRight": 8,
                },
                {
                  "color": "#3F434A",
                },
              ],
              {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          Ó§É
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "lineHeight": 18,
          },
          {
            "marginBottom": 8,
          },
          {
            "color": "#3F434A",
          },
        ]
      }
    >
      Mot de passe
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
          },
          {
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "overflow": "hidden",
          },
          {
            "borderBottomColor": "#3F434A",
            "borderLeftColor": "#3F434A",
            "borderRightColor": "#3F434A",
            "borderTopColor": "#3F434A",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="password"
          className="text-[16px] leading-[20px] font-[SpaceGrotesk-Regular] text-[14px] leading-[18px]"
          cursorColor="#6C464F"
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Votre mot de passe"
          secureTextEntry={true}
          selectionColor="#6C464F"
          testID="input/password"
          textContentType="password"
          value=""
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={5}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="input/password/RightIcon"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 24,
              },
              [
                {
                  "marginRight": 8,
                },
                {
                  "color": "#3F434A",
                },
              ],
              {
                "fontFamily": "IcoMoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          Ó§Å
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    event="clic_forgot_password"
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "marginTop": 12,
        },
        {
          "alignSelf": "flex-end",
        },
        [
          {
            "opacity": undefined,
          },
          undefined,
        ],
      ]
    }
    testID="forgotPassword/button"
  >
    <Text
      style={
        [
          {
            "fontFamily": "SpaceGrotesk-Medium",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#3F434A",
          },
          {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Mot de passe oubli√© ?
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animation="scale"
    collapsable={false}
    event="clic_sign_in"
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        {
          "backgroundColor": "#6C464F",
        },
        {
          "height": 44,
        },
        {
          "marginTop": 32,
        },
        [
          {
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
          undefined,
        ],
      ]
    }
    testID="signIn/button"
  >
    <ActivityIndicator
      color="white"
    />
  </View>
</View>
`;
